Implement `TypeInfo` for `Duration`

Something in Substrate was merged recently, probably https://github.com/paritytech/substrate/pull/13302, which made our code no longer compile with this error:

```
error[E0277]: the trait bound `std::time::Duration: scale_info::TypeInfo` is not satisfied
   --> crates/sp-consensus-subspace/src/lib.rs:479:1
    |
479 | / sp_api::decl_runtime_apis! {
480 | |     /// API necessary for block authorship with Subspace.
481 | |     pub trait SubspaceApi<RewardAddress: Encode + Decode> {
482 | |         /// The slot duration in milliseconds for Subspace.
...   |
534 | |     }
535 | | }
    | |_^ the trait `scale_info::TypeInfo` is not implemented for `std::time::Duration`
```

Encoding and decoding is supported for `Duration`, so `TypeInfo` should be too.

--- a/src/impls.rs
+++ b/src/impls.rs
@@ -31,6 +31,7 @@ use crate::prelude::{
     },
     string::String,
     sync::Arc,
+    time::Duration,
     vec::Vec,
 };
 
@@ -160,6 +161,24 @@ impl_for_non_zero!(
     NonZeroU64: u64,
     NonZeroU128: u128
 );
+
+impl TypeInfo for Duration {
+    type Identity = Self;
+
+    fn type_info() -> Type {
+        Type::builder().path(Path::prelude("Duration")).composite(
+            Fields::unnamed()
+                .field(|f| {
+                    // Seconds
+                    f.ty::<u64>().type_name("u64")
+                })
+                .field(|f| {
+                    // Nanoseconds
+                    f.ty::<u32>().type_name("u32")
+                }),
+        )
+    }
+}
 
 impl<T> TypeInfo for Vec<T>
 where

--- a/src/prelude.rs
+++ b/src/prelude.rs
@@ -38,6 +38,7 @@ cfg_if! {
             ops,
             string,
             sync,
+            time,
             vec,
         };
     } else {
@@ -60,6 +61,7 @@ cfg_if! {
             mem,
             num,
             ops,
+            time,
         };
     }
 }

